sphinx/application.py

```diff
@@ -26,6 +26,7 @@
 from docutils import nodes
 from docutils.parsers.rst import convert_directive_function, \
     directives, roles
+from pkg_resources import iter_entry_points
```

Comment:
I agree for depending to setuptools. In fact, most environments have setuptools and troubles derived from setuptools are less than before.
Only one thing, I hear that Ubuntu 16.04 is having trouble with setuptools and pkg_resources.
https://bugs.launchpad.net/ubuntu/+source/python-pip/+bug/1635463
Depending on setuptools, Sphinx may be affected by this bug. However, let's think about after the problem has occurred.